src/webnn_native/dml/GraphDML.cpp

```diff
@@ -888,7 +976,82 @@ namespace webnn_native::dml {
     }
 
     MaybeError Graph::AddConvTranspose2d(const op::ConvTranspose2d* convTranspose2d) {
-        return DAWN_UNIMPLEMENTED_ERROR("ConvTranspose2D has not been supported on DirectML.");
+        auto inputsOperand = convTranspose2d->Inputs();
+        DAWN_ASSERT(inputsOperand.size() == 2 || inputsOperand.size() == 3);
+        DAWN_ASSERT(mExpression.find(inputsOperand[0].Get()) != mExpression.end());
+        ::dml::Expression input = mExpression.at(inputsOperand[0].Get());
+        DAWN_ASSERT(mExpression.find(inputsOperand[1].Get()) != mExpression.end());
+        ::dml::Expression filter = mExpression.at(inputsOperand[1].Get());
+
+        const ConvTranspose2dOptions* options = convTranspose2d->GetOptions();
+        if (options->inputLayout == wnn::InputOperandLayout::Nhwc) {
+            input = ReinterpretInputLayout(NhwcToNchw, input);
+        }
+        if (options->filterLayout != wnn::ConvTranspose2dFilterOperandLayout::Iohw) {
+            filter = ReinterpretFilterLayoutAsIohw(options->filterLayout, filter);
+        }
+        ::dml::Span<const uint32_t> strides(reinterpret_cast<const uint32_t*>(options->strides),
+                                            options->stridesCount);
+        ::dml::Span<const uint32_t> dilations(reinterpret_cast<const uint32_t*>(options->dilations),
+                                              options->dilationsCount);
+        ::dml::Span<const uint32_t> outputPadding(
+            reinterpret_cast<const uint32_t*>(options->outputPadding), options->outputPaddingCount);
+
+        ::dml::TensorDimensions inputDims = input.GetOutputDesc().sizes;
+        ::dml::TensorDimensions filterDims = filter.GetOutputDesc().sizes;
+        auto padding = options->autoPad == wnn::AutoPad::Explicit
+                           ? ExplicitPadding<ConvTranspose2dOptions>(options)
+                           : ImplicitPaddingforConvTranpose2d(options, inputDims, filterDims);
+        // dml::Span just holds the refernces, need a variable to hold the memory.
+        std::vector<const uint32_t> startPaddingVector = {padding[0], padding[2]};
+        ::dml::Span<const uint32_t> startPadding(startPaddingVector);
+        std::vector<const uint32_t> endPaddingVector = {padding[1], padding[3]};
+        ::dml::Span<const uint32_t> endPadding(endPaddingVector);
+
+        ::dml::TensorDimensions outputShape = {};
+        outputShape.push_back(inputDims[0]);   // N
+        outputShape.push_back(filterDims[1]);  // C
+        if (options->outputSizes != nullptr) {
+            std::vector<int32_t> outputSizes(options->outputSizes,
+                                             options->outputSizes + options->outputSizesCount);
+            outputShape.push_back(outputSizes[0]);
```

Comment:
You can define the uint32_t outputHeight and uint32_t outputWidth outside, so you can assign them like 
```
if (options->outputSizes != nullptr) {
  outputHeight  = options->outputSizes[0];
  outputWidth  = options->outputSizes[1];
} else {
  outputHeight  = 
  outputWidth  = 
}
outputShape.push_back(outputHeight);
outputShape.push_back(outputWidth);
```

Reply:
Thanks for the suggestion, I will pay more attention to some such problems that the code can be optimized in the future. PTAL, @fujunwei.